### PR TITLE
Minor app service related changes

### PR DESCRIPTION
**Pull request by agr**
Renamed a variable and a method to emphasize their relation to cloud service infrastructure (not the cloud hosting in general).
Removed duplicate code that figured out machine name.
Added `<applicationInitialization>` section to the web.config.

### CODE REVIEW

File: src/NuGetGallery.Services/Configuration/ConfigurationService.cs [hunk 1]
```diff
@@ -23,7 +24,7 @@ public class ConfigurationService : IGalleryConfigurationService
         protected const string ServiceBusPrefix = "AzureServiceBus.";
         protected const string PackageDeletePrefix = "PackageDelete.";
 
-        private bool _notInCloud;
+        private bool _notInCloudService;
```
Comment (joelverhagen) [inline src/NuGetGallery.Services/Configuration/ConfigurationService.cs]: Hahahaha, this is an awesome delta.

Comment (agr) [reply src/NuGetGallery.Services/Configuration/ConfigurationService.cs]: Well, it's the actual meaning of the variable: if cloud service API is available. We would be able to drop it completely once we are off cloud services.

File: src/NuGetGallery/Web.config [hunk 2]
```diff
@@ -505,6 +505,10 @@
         </rule>
       </rules>
     </rewrite>
+    <applicationInitialization>
+      <add initializationPage="/"/>
+      <add initializationPage="/api/health-probe"/>
+    </applicationInitialization>
```
Comment (shishirx34) [inline src/NuGetGallery/Web.config]: who uses these?

Comment (agr) [reply src/NuGetGallery/Web.config]: First one is added mostly to align with the search service, and the second one is watched at least by the internal load balancer, yes.

**Pull request merged**